Support reversing with .isel

Adding 
```python
ds.isel(y=slice(None, None, -1))
```

to https://github.com/dcherian/rasterix/blob/d6242e173e691fb3a017c42dbeca1a77c72184fb/tests/test_raster_index.py#L31

fails  with

```
../../miniforge3/envs/rasterix/lib/python3.12/site-packages/xarray/core/indexes.py:2097: in _apply_indexes
    new_index = getattr(index, func)(index_args)
src/rasterix/raster_index.py:408: in isel
    return RasterIndex(new_indexes)
src/rasterix/raster_index.py:344: in __init__
    "x": self._wrapped_indexes["x"].axis_transform.size,
E   KeyError: 'x'
```

I suspect this is fixed on #31, but we should add a test

<img width="939" alt="Image" src="https://github.com/user-attachments/assets/cacbafda-53d4-47aa-8411-910460d11526" />

--- a/src/rasterix/raster_index.py
+++ b/src/rasterix/raster_index.py
@@ -201,14 +201,10 @@ class AxisAffineTransform(CoordinateTransform):
         return self.forward({self.dim: np.arange(self.size)})
 
     def slice(self, slice: slice) -> AxisAffineTransform:
-        start = max(slice.start or 0, 0)
-        stop = min(slice.stop or self.size, self.size)
-        step = slice.step or 1
-
-        # TODO: support reverse transform (i.e., start > stop)?
-        assert start < stop
-
-        size = (stop - start) // step
+        newrange = range(self.size)[slice]
+        start = newrange.start
+        step = newrange.step or 1
+        size = len(newrange)
         scale = float(step)
 
         if self.is_xaxis:

--- a/tests/test_raster_index.py
+++ b/tests/test_raster_index.py
@@ -69,6 +69,18 @@ def test_sel_slice():
     assert actual_transform == actual.rio.transform()
     assert actual_transform == (transform * Affine.translation(0, 3))
 
+    reverse = ds.isel(y=slice(None, None, -1))
+    assert_identical(reverse.y, ds.y[::-1])
+
+    reverse = ds.isel(y=slice(8, 5, -1))
+    assert_identical(reverse.y, ds.y[8:5:-1])
+
+    reverse = ds.isel(y=slice(8, None, -1))
+    assert_identical(reverse.y, ds.y[8::-1])
+
+    reverse = ds.isel(y=slice(None, 5, -1))
+    assert_identical(reverse.y, ds.y[:5:-1])
+
 
 def test_crs() -> None:
     index = RasterIndex.from_transform(Affine.identity(), width=12, height=10)

